From 862aeb78303c7eb8b82e73ade65b6cf4308d2e58 Mon Sep 17 00:00:00 2001
From: Peter Korsgaard <peter@korsgaard.com>
Date: Wed, 15 May 2024 16:59:02 +0200
Subject: [PATCH] package/libvncserver: correct WITH_THREADS=ON/OFF logic

Commit ee59023794f5540e5 (package/libvncserver: bump to version 0.9.12)
changes the package from autotools to cmake and accidently inverted the
threads logic.

The reason this was not noticed is that the build system verifies if threads
support works and otherwise disables it even when -DWITH_THREADS=ON is
passed, E.G.  for a nothread configuration:

cmake ... -DWITH_THREADS=ON ...
..
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found

Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
Signed-off-by: Yann E. MORIN <yann.morin.1998@free.fr>
(cherry picked from commit 7b3c0b19d4d587f52546ac92386e8a890022bb24)
Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 package/libvncserver/libvncserver.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/libvncserver/libvncserver.mk b/package/libvncserver/libvncserver.mk
index 3a15a82d0800..082f838f1229 100644
--- a/package/libvncserver/libvncserver.mk
+++ b/package/libvncserver/libvncserver.mk
@@ -22,7 +22,7 @@ LIBVNCSERVER_CONF_OPTS += \
 	-DWITH_SDL=OFF \
 	-DWITH_TESTS=OFF

-ifneq ($(BR2_TOOLCHAIN_HAS_THREADS),y)
+ifeq ($(BR2_TOOLCHAIN_HAS_THREADS),y)
 LIBVNCSERVER_CONF_OPTS += -DWITH_THREADS=ON
 else
 LIBVNCSERVER_CONF_OPTS += -DWITH_THREADS=OFF
